16/umbraco-cms/reference/content-delivery-api/custom-property-editors-support.md

```diff
@@ -86,38 +72,28 @@ public class MyCustomPickerValueConverter : PropertyValueConverterBase, IDeliver
 
     private DeliveryApiCustomPicker? BuildDeliveryApiCustomPicker(object inter, bool expanding)
     {
-        if (!_publishedSnapshotAccessor.TryGetPublishedSnapshot(out IPublishedSnapshot? publishedSnapshot) ||
-            publishedSnapshot?.Content is null)
+        if (!Guid.TryParse(inter as string, out Guid id))
         {
             return null;
         }
+        
+        var content = publishedContentCache.GetById(id);
 
-        if (!Guid.TryParse(inter as string, out Guid id))
+        if (content is null)
         {
             return null;
         }
-
+        
         return new DeliveryApiCustomPicker { Id = id };
     }
 }
-
-public class DeliveryApiCustomPicker
-{
-    public Guid Id { get; set; }
-
-    public DeliveryApiItemDetails? ItemDetails { get; set; }
-}
-
-public class DeliveryApiItemDetails
-{
-    public string? Name { get; set; }
-
-    public IApiContentRoute? Route { get; set; }
-}
 ```
-
 {% endcode %}
 
+{% hint style="info" %}
+Umbraco developers will need to inject `IPublishedContentCache`, `IPublishedMediaCache`, `IPublishedMemberCache` and `IPublishedContentTypeCache` dependencies individually, instead of injecting the `IPublishedSnapshotAccessor` as would have been done previously.
```

Comment:
```suggestion
Umbraco developers need to inject `IPublishedContentCache`, `IPublishedMediaCache`, `IPublishedMemberCache`, and `IPublishedContentTypeCache` individually, instead of injecting the `IPublishedSnapshotAccessor` as in previous versions.
```